### PR TITLE
fix(useSpeechRecognition): reactive speech language

### PR DESCRIPTION
**Pull request by JMaylor**
Basic attempt at #719
I will update the demo another day so this uses something more appropriate for the non-English languages. For now you can use the demo to see that the language does change.

### CODE REVIEW

File: packages/core/useSpeechRecognition/index.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export interface SpeechRecognitionOptions extends ConfigurableWindow {
    *
    * @default 'en-US'
    */
-  lang?: string
+  lang?: string | Ref
```
Comment (wheatjs) [inline packages/core/useSpeechRecognition/index.ts:27]: Use `MaybeRef<string>` instead

File: packages/core/useSpeechRecognition/index.ts [hunk 2]
```diff
@@ -69,12 +69,14 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = {}) {
 
     recognition.continuous = continuous
     recognition.interimResults = interimResults
-    recognition.lang = lang
+    recognition.lang = isRef(lang) ? lang.value : lang
```
Comment (wheatjs) [inline packages/core/useSpeechRecognition/index.ts:72]: Use `unref(lang)` instead

File: packages/core/useSpeechRecognition/index.ts [hunk 3]
```diff
@@ -94,6 +96,7 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = {}) {
 
     recognition.onend = () => {
       isListening.value = false
+      if (isRef(lang) && recognition) recognition.lang = lang.value
```
Comment (wheatjs) [inline packages/core/useSpeechRecognition/index.ts:99]:
```js
if (recognition)
  recognition.lang = unref(lang)
```

**Review by wheatjs — CHANGES_REQUESTED** [review]

File: packages/core/useSpeechRecognition/index.ts (hunk 3 above)
Comment (JMaylor) [reply packages/core/useSpeechRecognition/index.ts:99]: For this - we only want to update `recognition.lang` if a ref is passed in - no point doing this if a plain string is used?

Comment (wheatjs) [reply packages/core/useSpeechRecognition/index.ts:99]: 👍

File: packages/core/useSpeechRecognition/index.md [hunk 4]
```diff
@@ -34,3 +34,27 @@ The following shows the default values of the options, they will be directly pas
   continuous: true,
 }
 ```
+
+You can also bind the lang property to a ref object. Note that updating this ref value cannot take effect until the next time the recognition is stopped.
```
Comment (antfu) [inline packages/core/useSpeechRecognition/index.md:38]: Have arguments accepts ref is a common practice in VueUse, I don't think we need to explicitly document it here. (but thanks!)

File: packages/core/useSpeechRecognition/index.ts [hunk 5]
```diff
@@ -69,12 +69,14 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = {}) {
...
     recognition.continuous = continuous
     recognition.interimResults = interimResults
-    recognition.lang = lang
+    recognition.lang = unref(lang)
 
     recognition.onstart = () => {
       isFinal.value = false
     }
 
+    if (isRef(lang)) watch(lang, (lang) => { if (recognition && !isListening.value) recognition.lang = lang })
```
Comment (antfu) [inline packages/core/useSpeechRecognition/index.ts:78]: Can you format L78 and L99 a bit for better formatting?

**Pull request merged**